narwhal/executor/tests/consensus_integration_tests.rs

```diff
@@ -128,9 +132,10 @@ async fn test_recovery() {
 
         // we expect to have recovered all the certificates between the "last executed certificate" (included)
         // up to the "last committed certificate" (included).
-        assert_eq!(
-            consensus_output.len(),
-            (num_of_committed_certificates - last_executed_certificate_index) as usize
+        assert_eq!(consensus_output.len(), 1);
+        assert!(
+            consensus_output[0].len()
+                >= (num_of_committed_certificates - last_executed_certificate_index) as usize
```

Comment:
I would suggest changing this to :

```
        assert_eq!(consensus_output[0].len(),  consensus_index_counter as usize
```

and we probably need to refactor the test (maybe in follow up PR) to better capture the current recovery semantics.

**Before this PR**, we were effectively recovering _all the committed certificates_ starting from _the last executed certificate_

**After this PR** we'll recover _all the committed certificates_ since the last committed subdag/leader. So we'll always expect to replay at least all the certificates of the latest subdag.

as we discussed offline nothing wrong given SUI's idempotency , but we might want to make it clearer in the test it self